github/branches/test_branch_protection.py

```diff
@@ -0,0 +1,48 @@
+#!/usr/bin/env python3
+
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+from github.branches.validate_compliance import Criteria
+from typing import Any, Optional
+
+from .retrieve_github_data import BranchOfInterest, get_repo_branch_protections
+from .retrieve_github_data import repos_to_check
+from . import validate_compliance
+
+# from .conftest import repos_to_check
+
+
+import pytest
+
+
+def idfn(val: Any) -> Optional[str]:
```

Comment:
NB: Should this be alongside `repos_to_check()`? And called `get_repo_id` or similar?

Reply:
My thinking had been "`repos_to_check()`" is a pure GitHub thing. "`idfns=`" is a pytest thing.

I think I can move it to the "`BranchOfInterest`" class, as I did for "`Criteria`", unless that feels "too different".